app/uk/gov/hmrc/cdsreimbursementclaimstubs/controllers/DeclarationInfoRequestController.scala

```diff
@@ -0,0 +1,84 @@
+import cats.syntax.EitherSyntax
+import com.eclipsesource.schema.drafts.Version4
+import com.eclipsesource.schema.drafts.Version4._
+import com.eclipsesource.schema.{SchemaType, SchemaValidator}
+import com.google.inject.{Inject, Singleton}
+import play.api.libs.json.{JsValue, Json}
+import play.api.mvc.{Action, ControllerComponents}
+import play.twirl.api.Html
+import uk.gov.hmrc.cdsreimbursementclaimstubs.models.DeclarationErrorResponse._
+import uk.gov.hmrc.cdsreimbursementclaimstubs.utils.DeclarationInfoResponses._
+import uk.gov.hmrc.cdsreimbursementclaimstubs.utils.Logging
+import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
+
+import scala.io.Source
+
+@Singleton
+class DeclarationInfoRequestController @Inject()(cc: ControllerComponents) extends BackendController(cc) with Logging with EitherSyntax{
+  lazy val schemaToBeValidated: SchemaType = {
+    Json
+      .fromJson[SchemaType](
+        Json.parse(
+          Source
+            .fromInputStream(
+              this.getClass.getResourceAsStream("/resources/OverpaymentDeclarationDisplayRequest-Schema.json")
+            )
+            .mkString
+        )
+      )
+      .get
+  }
+
+
+  val getDeclaration: Action[JsValue] = Action(parse.json) { implicit request =>
+    val validator = SchemaValidator(Some(Version4))
+
+    (for {
+      _ <- Either.cond(request.method.toUpperCase == "POST", "POST", WrongMethod)
+      _ <- request.headers.get("Accept").map(_.toLowerCase == "application/json").toRight(WrongAcceptHeader)
+      _ <- request.headers.get("Content-Type").map(_.toLowerCase == "application/json").toRight(WrongContentTypeHeader)
+      _ <- request.headers.get("X-Correlation-ID").toRight(NoCorrelationIdHeader)
+      _ <- request.headers.get("X-Forwarded-Host").toRight(NoXForwaredHostHeader)
+      _ <- request.headers.get("Authorization").toRight(NoAuthorizationHeader)
+      _ <- validator.validate(schemaToBeValidated, request.body).asEither.leftMap(err => makeEis(err.toString(), BAD_REQUEST))
+      declarationId <- (request.body \ "overpaymentDeclarationDisplayRequest" \ "requestDetail" \ "declarationId")
+        .toEither.leftMap(_ => NoDeclarationId)
+    } yield declarationId.as[String]
+      ).fold(
+      error => {
+        logger.warn(s"Returning http status ${error.errorDetail.errorCode}: ${error}")
+        toErrorResponse(error)
+      },
+      declarationId => declarationId match {
+        case "57WAFResponse11111" => Forbidden(Html(WafError.value))
+        case "57TimeoutResponse1" => toErrorResponse(Timeout)
+        case "57CouldNotProcess1" => toErrorResponse(CouldNotBeProcessed)
+        case "57NoDeclarationF11" => toErrorResponse(NoDeclarationFound)
+        case "57NoSecurityDepos1" => toErrorResponse(NoSecurityDeposits)
+        case "57EMPTYRESPONSE111" => toOkResponse(emptyResponse)
+        case "57MINIMUMRESPONSE1" => toOkResponse(minimumResponse(declarationId))
+        case _ => toOkResponse(fullResponse(declarationId))
```

Comment:
Something I wanted to do but didn't get time in the last sprint is to make a markdown table in the README for the different APIs were are simulating. The table columns would be:

1. Key
2. EIS Http Status Code Response

This will help the QAs to pick the correct EORI or Declaration Id to generate the required HTTP response for the test purposes. So in summary two tables:

1. TPIO5
2. ACC14